### ISSUE TITLE
How to generate multiple outputs and find the best?

### ISSUE THREAD

**Issue by XuyangBai** [SimingYan/Hybrid_Relative_Pose#1]
Hi @EternityDS Thanks for your sharing. I have two questions regarding the multiple outputs. In my understanding, you choose the corresponding eigenvector of  top-`k` eigenvalues in the spectral matching submodule to recover `k` relative poses. Is my understanding correct? BTW, I find many functions named `Spectral_Matching` or `Spectral_Matching_M` or `Hybrid_Spectral_Matching_M `in your code, may I ask what's the difference and which one are you indeed using in your paper? 

And my second question is during testing, how can we decide which hypothesis is the best. In the paper, I found "In the presence of multiple outputs, we define the best match as the one that minimizes the rotation error." So it seems you have to know the ground truth pose first, then you can decide which one is the best? But if there is no ground truth pose available, is there any way to decide the best guess?

Best,
Xuyang.

**Comment by EternityDS** [comment]
Hi Xuyang,

Yes, you are correct. We used Hybrid_Spectral_Matching_M in our paper. Others are old versions. 
We choose the best candidate pose compared to the ground truth. It would be great if you have any idea to decide the best pose without supervision.

Best,
Siming

**Comment by XuyangBai** [comment]
Hi Siming,

Thanks for your reply. I am also trying to some way to decide the best pose without knowing the ground truth, but no good solution up to now.

Best,
Xuyang

**Comment by XuyangBai** [comment]
Hi Siming @EternityDS 

I have several questions regarding the construction of the consistency matrix. In the paper, you mentioned in the supplementary material that the five gamma are hyper-parameters to be learned from data. But I found you used some pre-defined value for them (e.g. 0.5236 for sigmaAngle1 and sigmaAngle1, 0.3 for sigmaDist), may I ask how did you decide the value of these hyperparameters? In my understanding, these sigmas decide the importance among different geometric quantities as well as the feature similarity, so what does these values represent? And as you have a loss term for spectral matching, then if these five sigmas are fixed, what will this loss optimize?

Best,
Xuyang

**Comment by SimingYan** [comment]
We didn't release the training code yet. Will release soon.
